cdk/kittyhawk/src/application/base.ts

```diff
@@ -6,11 +6,13 @@ import { Ingress, IngressProps } from "../ingress";
 import { Service } from "../service";
 import { ServiceAccount } from "../serviceaccount";
 
+interface PortExclusiveIngressProps extends Omit<IngressProps, "port"> {}
+
 export interface ApplicationProps {
   /**
    * Ingress configuration
    */
-  readonly ingress?: IngressProps;
+  readonly ingress?: PortExclusiveIngressProps;
```

Comment:
instead of naming the type you make, you just do something like this (unless you want to use the named type in other places)

```js
readonly ingress?: Omit<IngressProps, "port">;
```

Reply:
Oh good catch, thanks!